### ISSUE TITLE
std.fmt c specifier should print numbers as unicode codepoints

### ISSUE THREAD

**Issue by Vexu** [ziglang/zig#5564]
It makes no sense to have unicode character literals but no way to format them.
```zig
test "" {
    @import("std").debug.warn("{c}\n", .{'あ'});
    // error: Cannot print integer that is larger than 8 bits as a ascii
}
```
The `c` specifier should format integers as unicode codepoints and a new specifier should be added for formatting them as their literal bytes.

This is how the [Go](https://golang.org/pkg/fmt/), [Python](https://docs.python.org/3/library/string.html#format-specification-mini-language), [Odin](https://github.com/odin-lang/Odin/blob/f992e36f9a620c3512d28658ed9b2e4b6fd3a730/core/fmt/fmt.odin#L717) and [Java](https://docs.oracle.com/javase/7/docs/api/java/util/Formatter.html) handle `c`.

**Comment by data-man** [comment]
In #3970 I proposed 'u'.
> `u`: output integer as an UTF-8 sequence. Integer type must have 21 bits at max.

But I prefer `U+FFFD` for invalid codepoints.

https://en.wikipedia.org/wiki/UTF-8#Invalid_byte_sequences
> The standard also recommends replacing each error with the replacement character "�" (U+FFFD).

**Comment by andrewrk** [comment]
Implemented by @LemonBoy in #6390, landed in 473cb1fd74d6d478bb3d5fda4707ce3f6e6e5bf6.
